phoenix-core/src/main/java/org/apache/phoenix/schema/types/PhoenixArray.java

```diff
@@ -561,6 +666,23 @@ public int estimateByteSize(int pos) {
 		}
 		
 		@Override
+		public String toString() {
+			StringBuilder sb = new StringBuilder(TO_STRING_BEGIN);
+			boolean isFirst = true;
+			final Format defaultFormat = null;
+			for (Byte o : byteArr) {
+				if (isFirst) {
+					isFirst = false;
+				} else {
+					sb.append(TO_STRING_SEPARATOR);
+				}
+				sb.append(this.baseType.toStringLiteral(o, defaultFormat));
```

Comment:
Good suggestion on toStringLiteral().

On the second comment.. I started to use PhoenixArray.toString() but that doesn't work. Arrays in primitive array subclasses contain primitives and there's an exception if 
for(Object o : arr) { ... } is used. If arr consists of primitives (e.g. ints), then for(Integer o : ints) works because type boxing/unboxing automatically does the right cast.

I guess another way would be to have a method that converts from primitive to object. Eg. int to Integer which would make toString generic.